File: backend/internal/entity/restaurant.go
package entity

import "gorm.io/gorm"

type Restaurant struct {
	gorm.Model
	Name                 string  `json:"name" gorm:"not null" binding:"required"`
	BranchName           string  `json:"branch_name" gorm:"size:120;default:'สาขาหลัก'"`
	RestaurantType       string  `json:"restaurant_type" gorm:"size:80;default:'ร้านอาหาร'"`
	Address              string  `json:"address"`
	Phone                string  `json:"phone"`
	Logo                 string  `json:"logo"`
	OpenTime             string  `json:"open_time" gorm:"size:5;default:'17:00'"`
	CloseTime            string  `json:"close_time" gorm:"size:5;default:'00:00'"`
	TableCount           int     `json:"table_count" gorm:"not null;default:12;check:restaurant_table_count_nonnegative,table_count >= 0"`
	ServiceChargeEnabled bool    `json:"service_charge_enabled" gorm:"not null;default:false"`
	ServiceChargeRate    float64 `json:"service_charge_rate" gorm:"type:numeric(7,4);not null;default:10;check:restaurant_service_charge_rate_range,service_charge_rate >= 0 AND service_charge_rate <= 100"`
	VATEnabled           bool    `json:"vat_enabled" gorm:"not null;default:false"`
	VATRate              float64 `json:"vat_rate" gorm:"type:numeric(7,4);not null;default:7;check:restaurant_vat_rate_range,vat_rate >= 0 AND vat_rate <= 100"`
	PromptPayName        string  `json:"promptpay_name"`
	PromptPayQRImage     string  `json:"promptpay_qr_image"`
	CoverImage           string  `json:"cover_image"`

	// Geofence for QR table ordering. A customer must be physically near the
	// restaurant to send an order straight to the kitchen. OrderRadiusMeters = 0
	// (or missing coordinates) disables the check entirely.
	Latitude          *float64 `json:"latitude" gorm:"type:double precision"`
	Longitude         *float64 `json:"longitude" gorm:"type:double precision"`
	OrderRadiusMeters int      `json:"order_radius_meters" gorm:"not null;default:0"`

	OwnerID uint  `json:"owner_id" gorm:"not null"`
	Owner   *User `json:"owner,omitempty" gorm:"foreignKey:OwnerID"`
}

File: mobile/src/types/restaurant.ts
import type { User } from './auth';

export interface Role {
  ID: number;
  restaurant_id?: number | null;
  name: string;
  display_name: string;
  permissions: string;
  is_system: boolean;
}

export interface MembershipUserSummary {
  ID: number;
  email: string;
  first_name: string;
  last_name: string;
  nickname: string;
  profile_image: string;
}

export interface Restaurant {
  ID: number;
  name: string;
  branch_name: string;
  restaurant_type: string;
  address: string;
  phone: string;
  logo: string;
  open_time: string;
  close_time: string;
  table_count: number;
  service_charge_enabled: boolean;
  service_charge_rate: number;
  vat_enabled: boolean;
  vat_rate: number;
  promptpay_name: string;
  promptpay_qr_image: string;
  cover_image: string;
  latitude?: number | null;
  longitude?: number | null;
  order_radius_meters?: number;
  owner_id: number;
  owner?: User;
  CreatedAt?: string;
  UpdatedAt?: string;
}

export type MembershipStatus = 'active' | 'suspended' | 'removed';

export interface Membership {
  ID: number;
  user_id: number;
  restaurant_id: number;
  role_id: number;
  permissions_override?: string | null;
  status: MembershipStatus;
  joined_at: string;
  invited_by_user_id?: number | null;
  user?: MembershipUserSummary;
  restaurant?: Restaurant;
  role?: Role;
}

export type InvitationStatus = 'pending' | 'accepted' | 'revoked' | 'expired';

export interface InvitationRestaurantSummary {
  ID: number;
  name: string;
  branch_name: string;
  address: string;
  logo: string;
}

export interface InvitationRoleSummary {
  ID: number;
  restaurant_id?: number | null;
  name: string;
  display_name: string;
  is_system: boolean;
}

export interface Invitation {
  ID: number;
  restaurant_id: number;
  role_id: number;
  email: string;
  expires_at?: string | null;
  status: InvitationStatus;
  accepted_at?: string | null;
  restaurant?: InvitationRestaurantSummary;
  role?: InvitationRoleSummary;
}

export interface AdminInvitation extends Invitation {
  token: string;
  invited_by_user_id: number;
  accepted_by_user_id?: number | null;
}

export interface RestaurantAuditLog {
  ID: number;
  restaurant_id: number;
  actor_user_id?: number | null;
  target_user_id?: number | null;
  invitation_id?: number | null;
  action: string;
  details: string;
  CreatedAt?: string;
  UpdatedAt?: string;
  actor_user?: User;
  target_user?: User;
  invitation?: Invitation;
}

export interface RestaurantInput {
  name: string;
  branch_name: string;
  restaurant_type: string;
  address?: string;
  phone?: string;
  logo?: string;
  open_time?: string;
  close_time?: string;
  table_count?: number;
  service_charge_enabled?: boolean;
  service_charge_rate?: number;
  vat_enabled?: boolean;
  vat_rate?: number;
  promptpay_name?: string;
  promptpay_qr_image?: string;
  cover_image?: string;
  split_zones?: boolean;
  latitude?: number | null;
  longitude?: number | null;
  order_radius_meters?: number;
}
